### PR TITLE
Avoid compute corrections twice

### PR DESCRIPTION
**Pull request by vbrancat**
This PR proposes an attempt to solve #92 and avoids to compute the correction layers twice.

At the moment, the correction layers are computed 1) to feed them to the geocodeSlc correction module 2) to allocate them inside the CSLC-S1 data product.

In this PR, the corrections are computed once, saved on disk and read when it is time to allocate them in the dataset.
May be there is a smarter way to do this but this PR gives a first stab to the issue.

### CODE REVIEW

**Comment by vbrancat** [comment]
@LiangJYu would you have time to look into this PR? Thanks

File: src/compass/utils/helpers.py [hunk 1]
```diff
@@ -318,3 +318,59 @@ def burst_bboxes_from_db(burst_ids, burst_db_file=None, burst_db_conn=None):
...
+                 data_type=gdal.GDT_Float32, format='ENVI'):
+    '''
+    Write a multiband GDAL-friendly raster to disk.
+    Each dataset allocated in the output file contains
+    a description of the dataset allocated for that band
+
+    Parameters
+    ----------
+    filename: str
+        Filepath where to store output dataset
```
Comment (LiangJYu) [inline src/compass/utils/helpers.py:333]:
```suggestion
        File path where to store output dataset
```
spelling nit

File: src/compass/utils/helpers.py [hunk 2]
```diff
@@ -318,3 +318,59 @@ def burst_bboxes_from_db(burst_ids, burst_db_file=None, burst_db_conn=None):
...
+    data_list: list, np.ndarray
```
Comment (LiangJYu) [inline src/compass/utils/helpers.py:334]:
```suggestion
    data_list: list of np.ndarray
```
Change for clarity - `,` separating the two items gives the wrong impression that it's either or.

File: src/compass/utils/helpers.py [hunk 3]
```diff
@@ -318,3 +318,59 @@ def burst_bboxes_from_db(burst_ids, burst_db_file=None, burst_db_conn=None):
...
+        List of numpy.ndarray to allocate for each
+        raster band. All datasets within the list
+        are assumed to have the same shape
+    descriptions: list, str
```
Comment (LiangJYu) [inline src/compass/utils/helpers.py:338]:
```suggestion
    descriptions: list of str
```
same as above

**Review by LiangJYu — APPROVED** [review]
LGTM - only formatting and documentation comments

File: src/compass/utils/helpers.py (hunk 2 above)
Comment (LiangJYu) [reply src/compass/utils/helpers.py:334]:
```suggestion
    data_list: list[np.ndarray]
```
This also works and I think is convention we've been using in COMPASS.

**Comment by vbrancat** [comment]
@seongsujeong Would you mind also having a look at this PR?

**Review by seongsujeong — APPROVED** [review]
LGTM

**Pull request merged**